src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -71,10 +82,23 @@ class PropertyAccessor implements PropertyAccessorInterface
     /**
      * Should not be used by application code. Use
      * {@link PropertyAccess::createPropertyAccessor()} instead.
+     *
+     * @param int $magicMethods A bitwise combination of the MAGIC_* constants
+     *                          to specify the allowed magic methods (__get, __set, __call)
+     *                          or self::DISALLOW_MAGIC_METHODS for none
      */
-    public function __construct(bool $magicCall = false, bool $throwExceptionOnInvalidIndex = false, CacheItemPoolInterface $cacheItemPool = null, bool $throwExceptionOnInvalidPropertyPath = true)
+    public function __construct(/*int */$magicMethods = self::MAGIC_GET | self::MAGIC_SET, bool $throwExceptionOnInvalidIndex = false, CacheItemPoolInterface $cacheItemPool = null, bool $throwExceptionOnInvalidPropertyPath = true)
     {
-        $this->magicCall = $magicCall;
+        if (\is_bool($magicMethods)) {
```

Comment:
Wouldn't it make sense to turn the `$throwExceptionOnInvalidIndex` and `$throwExceptionOnInvalidPropertyPath` into flags too?

Reply:
I still think we should see https://github.com/symfony/symfony/issues/31126#issuecomment-484362754

Reply:
I agree as well about this, but I'd rather provide a dedicated PR. Anyway we have the issue to not loose track of it. :)